webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSIntegrationProcessorJAXWS_JMS.java

```diff
@@ -92,37 +92,40 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         }
 
         //extract SOAP-over-JMS 1.0 bindings
-        List<JMSEndpointMetaData> list = new LinkedList<JMSEndpointMetaData>();
-        if (!map.isEmpty()) {
-
-            for (String wsdlLocation : map.keySet()) {
-                try {
-                    final ResourceRoot resourceRoot = getWsdlResourceRoot(unit, wsdlLocation);
-                    if (resourceRoot == null) continue;
-                    final VirtualFile wsdlLocationFile = resourceRoot.getRoot().getChild(wsdlLocation);
-                    final URL url = wsdlLocationFile.toURL();
-                    SOAPAddressWSDLParser parser = new SOAPAddressWSDLParser(url);
-                    for (AnnotationInstance ai : map.get(wsdlLocation)) {
-                        String port = ai.value(PORT_NAME).asString();
-                        String service = ai.value(SERVICE_NAME).asString();
-                        AnnotationValue targetNS = ai.value(TARGET_NAMESPACE);
-                        String tns = targetNS != null ? targetNS.asString() : null;
-                        QName serviceName = new QName(tns, service);
-                        QName portName = new QName(tns, port);
-                        String soapAddress = parser.filterSoapAddress(serviceName, portName, SOAPAddressWSDLParser.SOAP_OVER_JMS_NS);
-                        if (soapAddress != null) {
-                            ClassInfo webServiceClassInfo = (ClassInfo) ai.target();
-                            String beanClassName = webServiceClassInfo.name().toString();
-                            //service name ?
-                            list.add(new JMSEndpointMetaData(beanClassName, port, beanClassName, wsdlLocation, soapAddress));
-                        }
+        List<JMSEndpointMetaData> list = new LinkedList<>();
```

Comment:
May be you can safe some resource here:
L94:  `List<JMSEndpointMetaData> list = Collections.emptyList();`
L95: keep the guard: `if (!map.isEmpty()) {`
L96: `list = new LinkedList<>();`

Reply:
I was thinking the same (existing early with an empty list if the map is empty), but the list gets passed into the constructor of `JMSEndpointsMetaData`, and that will replace any empty list with `Collections.emptyList()`. So I assume any unnecessary list will be garbage-collected away pretty quickly.

But I'm happy to change it if you feel it's worthwhile (saves other readers from needing to check how the list is used I guess).

Reply:
@zodac The previous implementation had the guard and the receiver of the list `JMSEndpointsMetaData` checks for null and emptiness. I assume this is intentionally and avoiding unnecessary object creation/processing seems useful here. `Collections.emptyList()` returns a static object - no (unnecessary!) garbage-collection will happen.

Reply:
Sounds fair. I'll extract the loop logic to another method to make it clearer to return early with an empty list instead.